### PR TITLE
feat: display due tasks on the dashboard

### PR DESCRIPTION
**Pull request by djaiss**
image

### CODE REVIEW

File: domains/Contact/ManageTasks/Web/Controllers/ContactModuleTaskController.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ public function update(Request $request, int $vaultId, int $contactId, int $task
             'contact_task_id' => $taskId,
             'label' => $request->input('label'),
             'description' => null,
-            'due_at' => $dueAt,
+            'due_at' => $dueAt == '' ? null : $dueAt,
```
Comment (asbiin) [inline domains/Contact/ManageTasks/Web/Controllers/ContactModuleTaskController.php]: should be
```suggestion
            'due_at' => $dueAt === '' ? null : $dueAt,
```

File: domains/Vault/ManageVault/Web/ViewHelpers/VaultShowViewHelper.php [hunk 2]
```diff
@@ -127,4 +128,49 @@ public static function favorites(Vault $vault, User $user): Collection
...
             });
     }
+
+    public static function dueTasks(Vault $vault, User $user): Collection
+    {
+        $contactIds = $vault->contacts()->select('id')->get()->toArray();
+        $tasks = DB::table('contact_tasks')
+            ->where('completed', false)
+            ->whereIn('contact_id', $contactIds)
+            ->where('due_at', '<=', Carbon::now()->copy()->addDays(30))
```
Comment (asbiin) [inline domains/Vault/ManageVault/Web/ViewHelpers/VaultShowViewHelper.php]: Why do you need the `copy()` here?

**Review by asbiin — CHANGES_REQUESTED** [review]

File: domains/Vault/ManageVault/Web/ViewHelpers/VaultShowViewHelper.php (hunk 2 above)
Comment (djaiss) [reply domains/Vault/ManageVault/Web/ViewHelpers/VaultShowViewHelper.php]: Sometimes it fucks things up if we don't do that, but you are right, it's probably not needed.

Comment (djaiss) [reply domains/Vault/ManageVault/Web/ViewHelpers/VaultShowViewHelper.php]: fixed

**Pull request merged**